### PR TITLE
Remove the circular dependency between `diktat-rules` and `diktat-ruleset`

### PR DESCRIPTION
**Pull request by 0x6675636b796f75676974687562**
### What's done:

 * Now it's possible to run `mvn clean install` w/o skipping tests.
 * Reverted ad66a04da79f66db5c452197ece937f666ed14c9 (was useless in the 1st
   place).
 * Reverted a67811df0a6dcb27e047fdd35f09ae4478fc0a41 (no longer necessary).
 * Partially reverted 54c8ccefc3009b8dd2a493c30590168318765b89 (no longer
   necessary).
 * See #1478.

### CODE REVIEW

File: diktat-rules/pom.xml [hunk 1]
```diff
@@ -161,6 +161,73 @@
...
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- Create a fat JAR to be used in smoke tests, removing the
+                 circular dependency between `diktat-rules` and `diktat-ruleset`. -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
```
Comment (petertrr) [inline diktat-rules/pom.xml:190]: Why do we need a different plugin here? In `diktat-ruleset` we use `assembly-plugin`.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -49,16 +49,10 @@ jobs:
...
-        run: mvn -B clean install
-        # "diktat-ruleset" is a module that creates a fat JAR, which is,
-        # obviously, never released. Since we *do need* this fat JAR in order to
-        # run smoke tests, we *skip tests entirely* in this step.
-      - name: Deploy artifacts (skipping the clean, test, and verify phases)
-        run: mvn -B deploy -Prelease --projects "!diktat-ruleset" -DskipTests=true
+      - name: Deploy artifacts
+        run: mvn -B clean deploy -Prelease --projects '!diktat-ruleset'
+      - name: Build diktat.jar
+        run: mvn -B package --projects diktat-ruleset
```
Comment (petertrr) [inline .github/workflows/release.yml:55]: I thought that we would get rid of `diktat-ruleset` completely and upload fat jar from `diktat-rules`. What do you think about it?

File: diktat-rules/pom.xml (hunk 1 above)
Comment (0x6675636b796f75676974687562) [reply diktat-rules/pom.xml:190]: In `diktat-ruleset`, we use the `assmebly-plugin` with the pre-configured assembly profile, and the resulting fat JAR is the **only** artifact that module produces (as you remember, the execution of `jar-plugin` is suppressed).

In `diktat-rules`, the situation is different: we still need the `jar-plugin` to run and produce a regular JAR, and we also need the fat JAR to be **excluded** from the list of module artifacts (so that this fat JAR here is not installed/deployed/used as a dependency).

That's why the `assembly-plugin` doesn't seem suitable here (not mentioning it would reauire a separate XML configuration in an external XML file).

Basically, the task was a bit more complex than just copying a plug-in configuration from one module to another.

File: .github/workflows/release.yml (hunk 2 above)
Comment (0x6675636b796f75676974687562) [reply .github/workflows/release.yml:55]: I've observed problems producing **two** JAR artifacts from a single module: `diktat-maven-plugin` fails to build for whatever reason, see https://github.com/saveourtool/diktat/blob/infra/andrey.shcheglov/misc/diktat-rules/pom.xml#L186.

Additionally, to preserve compatibility, for the fat JAR we need to provide a valid `groupId` and `artifactId` (different from those of `diktat-rules`), craft a custom `pom.xml`, etc. This is a way more general task than simply resolving a circular dependency b/w modules, with little profit at the end.

**Review by petertrr — APPROVED** [review]
